### PR TITLE
Add EIP-5792 support to Safe Apps Provider

### PR DESCRIPTION
**Pull request by iamacook**
## Summary

With [EIP-5792](https://eips.ethereum.org/EIPS/eip-5792), transaction batching is possible. As it is also possible to batch transactions with our Safe Apps SDK, this includes the relevant JSON-RPC methods to propagate the feature into the Safe Apps Provider.

This implementation has been tested with the experimental implementation of [EIP-5792](https://viem.sh/experimental/eip5792/client) in viem.

Note: this takes inspiration from [the `safe-wallet-web` EIP-5792 implementation](https://github.com/safe-global/safe-wallet-web/blob/dev/src/services/safe-wallet-provider/index.ts).

### `wallet_sendCalls`

This method is responsible for dispatching the batch. Our implementation maps the incoming "calls" to the relevant `sdk.txs.send` call, which creates a `multiSend` call in our interface.

The method should return an ID for fetching the status of the calls. In our case, this is the `safeTxHash` of the `multiSend` transaction.

### `wallet_getCallsStatus`

Accepting the aforementioned `id` (`safeTxHash`), this returns the status and receipt(s) of the "calls". By using the `safeTxHash`, we fetch the details from the Transaction Service and transaction receipt, mapping the responses to the expected data structure.

### `wallet_showCallsStatus`

This should open the "status" in the wallet UI. As our SDK cannot navigate the UI, this is marked as unsupported. (For reference, see [the `safe-wallet-web` implementation](https://github.com/safe-global/safe-wallet-web/blob/dev/src/services/safe-wallet-provider/useSafeWalletProvider.tsx#L185-L193) - navigating to the transaction details.)

### `wallet_getCapabilities`

Responsible for returning the "features" of the wallet, this returns the support for `atomicBatch` (batching of transactions) for the current Safe.

## Changes

- Extend the Safe Apps Provider to support the following methods:
  - `wallet_sendCalls`
  - `wallet_getCallsStatus`
  - `wallet_getCapabilities`
- Explicitly mark the following methods as unsupported:
  - `wallet_showCallsStatus`
- Migrate number to hex conversion to use a new helper

### CODE REVIEW

File: packages/safe-apps-provider/src/provider.ts [hunk 1]
```diff
@@ -195,6 +195,90 @@ export class SafeAppProvider extends EventEmitter implements EIP1193Provider {
...
 
+      case 'wallet_sendCalls': {
+        if (params[0].from !== this.safe.safeAddress) {
+          throw Error('Invalid from address');
+        }
+
+        const txs = params[0].calls.map(
+          (call: { to?: `0x${string}`; data?: `0x${string}`; value?: `0x${string}`; chainId?: `0x${string}` }) => {
+            if (call.chainId !== numberToHex(this.chainId) || !call.to) {
+              throw new Error('Invalid call');
```
Comment (schmanu) [inline packages/safe-apps-provider/src/provider.ts:206]: We should maybe give more details: e.g. "Invalid calldata: The Safe is not available on the chain"

**Review by schmanu — APPROVED** [review]
Looks correct. We should maybe update the Test Safe app to easily test this from a Safe App.

File: packages/safe-apps-provider/src/provider.ts (hunk 1 above)
Comment (iamacook) [reply packages/safe-apps-provider/src/provider.ts]: I improved the error messages in c3ab8181e8a18c52b1f5fbaa13f78e1e5ec0b292. What do you think?

**Comment by iamacook** [comment]
> We should maybe update the Test Safe app to easily test this from a Safe App.

I agree. Considering it will require quite some changes (and that it is not release-dependant), I'd suggest doing it in a follow up PR. What do you think?

**Review by schmanu — APPROVED** [review]

**Review by dasanra — APPROVED** [review]

**Pull request merged**